tests/unit/database_cache_backend_test.py

```diff
@@ -0,0 +1,35 @@
+import time
+
+import pytest
+
+from caching.backends import DatabaseBackend
+
+
+@pytest.mark.asyncio
+async def test_backend(tmpdir):
+    backend = DatabaseBackend(url=f"sqlite:///{tmpdir}/test.db")
+    await backend.setup()
+
+    assert await backend.get("foo") is None
+    await backend.set("foo", "bar", 100)
+    assert await backend.get("foo") == "bar"
+    assert await backend.get_with_ttl("foo") == (100, "bar")
+    time.sleep(3)
+    assert await backend.get_with_ttl("foo") == (97, "bar")
+
+
+@pytest.mark.asyncio
+async def test_backend_postgres():
+    # uses the configuration from docker-compose.yaml db container
+    backend = DatabaseBackend(url="postgresql://postgres:postgres@localhost/storage")
+    try:
+        await backend.setup()
+    except OSError as e:
+        print(e)
+        pytest.skip("Failed to connect to postgres, probably container is not running. Skipping test")
```

Comment:
So basically, every time the backend cannot be set up due to OSError, this test is being skipped? A nice functionality, by the way, but this could also hide corner/edge cases. Since you only catch OSError, this should be fine, though :)

Reply:
I agree, it is mainly thought for the github actions. We could start using some python docker test utilities, where the required containers are started by the unit test framework. But that seems a little overkill just for postgres - 
For splash and lighthouse we can simply mock them which make the whole test setup a lot faster and less fragile

Reply:
Alright, sounds good :) Should I leave this open for now, or would splash and lighthouse be mocked in the future.

Reply:
Splash and lighthouse are already mocked in the integration tests :-)